webapp/channels/src/components/channel_view/channel_view.tsx

```diff
@@ -149,7 +149,7 @@ export default class ChannelView extends React.PureComponent<Props, State> {
                     </div>
                 </div>
             );
-        } else {
+        } else if (this.props.channelRoleLoaded) {
```

Comment:
just an observation, should we use !channelRoleLoaded and have fallback to the regular createPost. But i think it will not solve our problem wont it?

Reply:
I don't completely get what you mean. In order to have everything "together", I wanted to do `} else if (!this.props.channelRoleLoaded) {/* no roles component */} else {/* postInput */}` but the linter was complaining. But I guess I could have just created a new variable as `const showRoleWarning = !this.props.channelRoleLoaded;` and use that variable instead...

But still, not sure if that is what you are referring to.

Reply:
It does seem nice if the happy path case is at the end. We could also flip the name of the prop from `channelRoleLoaded` to `channelRolesLoading` as well